bundles/org.openhab.binding.magentatv/README.md

```diff
@@ -0,0 +1,237 @@
+You could send keys as you press them on the remote and the binding receives program information when the channel is switched (no
+The binding provides device discovery, sending keys for the remote and also receiving program information/events.
+
+## Supported Models
+
+* Deutsche Telekom Media Receiver MR401B - fully supported
+* Deutsche Telekom Media Receiver MR201  - fully supported
+* Deutsche Telekom Media Receiver MR400  - supported with minor restrictions (POWER key)
+* Deutsche Telekom Media Receiver MR200  - supported with minor restrictions (POWER key)
+* Deutsche Telekom Media Receiver MR3xx  - NOT supported (different platform)
+* Deutsche Telekom Media Receiver MR1xx  - NOT supported (different platform)
+
+
+### Receiver Standby Mode
+
+The Media receiver has 3 different standby modes, which could be selected in the receiver's settings.
+
+- Standby - full standby, receiver active all the time
+- Suspend/Resume - The receiver goes to sleep mode, but could be awaked by a Wake-on-LAN packet
+- Shutdown - Powering off will shutdown the receiver, can be awaked only with the power button
+
+Standby provides the best results, because the binding could wake-up the receiver (Power On/Off). Suspend/Resume would require a Wake-on-LAN, which could be done, but is currently not implemented. 
+Shutdown turns the receiver off, which requires a manual PowerOn.
+
+There is no way to detect the "display status" of the receiver. 
+The binding detects Power-Off with the MR401B/MR201 by listening to UPnP events, but can't verify the status when started.
+You need to take care on the current status if you power on/off the receiver from scenes.
+Check the current status before sending the POWER button, because POWER is a toggle, not ON or OFF (see sample rules).
+
+### Discovery
+
+The auto discovery starts when the binding is loaded.
+UPnP will be used to discover receivers on the local network.
+Based on UPnP all necessary parameters are auto-discovered.
+The receiver needs to be powered on to get discovered.
+
+### Supported Things
+
+|Thing    |Type                                                           |
+|---------|-------------------------------------------------------------- |
+|receiver |A MagentaTV Receiver, the binding supports multiple receivers. |
+
+
+
+# Device setup
+
+Once the thing will be added from the Inbox in PaperUI you'll need your T-Online credentials to query the userID.
+Open the thing configuration and enter the credentials:
+
+|Field            |Description                                      |
+|---------------- |------------------------------------------------ |
+|Account Name     |Your T-Online user id, e.g. test7017@t-online.de |
+|Account Password |The password for your Telekom account.           |
+
+For security reasons the credentials will be automatically deleted from the thing configuration (replaced with '***' in the thing config) after the initial authentication process.
```

Comment:
There might be a much more elegant solution for that: Create a console command that does the login and returns the parameter that is required for the thing configuration (what exactly is it?).
Check the Tesla binding (https://www.openhab.org/addons/bindings/tesla/#bridge-configuration), which offers this way (additionally to putting the credentials directly in the Thing configuration).

Reply:
who wants to use the console?

Reply:
Who wants to enter his password in a web UI, which is not related to the service the password belongs to? That's against any security best practices.

Reply:
@kai why does it not belong to the service? In fact that's a usual OAuth flow, which provides a token that is required for pairing. If credentials should never stored in textual thing configs we need to change a lot of bindings. What do you request?

Reply:
refactored, see comment below